src/index.js

```diff
@@ -287,17 +287,47 @@ export default function (babel) {
     return (typeof node.skinny === "boolean");
   }
 
+  // c/p babel-traverse/src/path/family.js getCompletionRecords
+  function getTailExpressions(path, allowLoops) {
+    let paths = [];
+
+    const add = function add(_path) {
+      if (_path) paths = paths.concat(getTailExpressions(_path));
+    };
+
+    if (path.isIfStatement()) {
+      add(path.get("consequent"));
+      add(path.get("alternate"));
+    } else if (path.isDoExpression()) {
+      add(path.get("body"));
+    } else if (allowLoops && (path.isFor() || path.isWhile())) {
```

Comment:
Nope; since a loop is a statement, it's not processed in `addImplicitReturns`; only `ExpressionStatement`s, `VariableDeclaration`s, and `FunctionDeclaration`s are.